manifests/deploy.pp

```diff
@@ -27,6 +27,7 @@
   $servergroups = hiera('jboss::deploy::servergroups', undef),
   $controller   = $::jboss::controller,
   $runasdomain  = $::jboss::runasdomain,
+  $runtime_name = undef,
```

Comment:
I think that `$runtime_name` shoul be evaluated to something (best to name of archive) so it is really optional. Now, it is required because if you not passed it, code will fail.